File: 0801-0900/0804.unique-morse-code-words.py
class Solution(object):
    MS = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
    def uniqueMorseRepresentations(self, words):
        """
        :type words: List[str]
        :rtype: int
        """
        def convert(word):
            return ''.join(Solution.MS[ord(c) - ord('a')] for c in word)
        return len(set(map(convert, words)))

File: 0801-0900/0804-unique-morse-code-words.cpp
const string S[26] = {".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};

class Solution {
public:
    int uniqueMorseRepresentations(vector<string>& words) {
        unordered_set<string> se;
        for (const string &s : words) {
            string buf;
            for (char c : s) {
                buf += S[c - 'a'];
            }
            se.insert(buf);
        }
        return se.size();
    }
};
